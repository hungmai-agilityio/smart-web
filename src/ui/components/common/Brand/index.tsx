import Link from 'next/link';
import Image from 'next/image';

interface BrandProps {
  logo: string;
  url?: string;
}

export const Brand = ({ logo, url = '/' }: BrandProps) => {
  return (
    <Link href={url} className="cursor-pointer">
      <Image src={logo} alt="logo" width={160} height={84} priority />
    </Link>
  );
};
