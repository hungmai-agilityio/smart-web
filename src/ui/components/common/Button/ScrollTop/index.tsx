'use client';

import Image from 'next/image';

export const ScrollTopButton = () => {
  const scrollTop = () => {
    window.scrollTo({ top: 0, behavior: 'smooth' });
  };

  return (
    <div className="flex justify-center cursor-pointer" onClick={scrollTop}>
      <Image
        src="/go-top-button.svg"
        alt="Go top icon"
        width={36}
        height={36}
        priority
      />
    </div>
  );
};
