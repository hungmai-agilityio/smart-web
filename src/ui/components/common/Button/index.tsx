'use client';

import { ButtonHTMLAttributes } from 'react';
import { clsx } from 'clsx';

// Constants
import { TYPE } from '@/constants';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: TYPE.PRIMARY | TYPE.SECOND;
  styles?: string;
}

export const Button = ({ name, variant, styles, disabled }: ButtonProps) => {
  const typeClasses = {
    [TYPE.PRIMARY]: 'bg-second',
    [TYPE.SECOND]:
      'bg-white border-primary border-2 border-primary hover:bg-second hover:border-0'
  };

  const typeClass = typeClasses[variant || TYPE.PRIMARY];

  const handleClick = () => {
    alert('We will update feature later');
  };

  return (
    <button
      className={clsx(
        'text-primary px-2 rounded-full font-light truncate',
        'w-24 h-6 text-xs sm:w-28 sm:h-8 sm:text-sm md:w-36 md:h-10 md:text-base lg:w-52 lg:h-12 lg:text-lg',
       disabled && 'cursor-not-allowed hover:bg-gray-400',
        typeClass,
        styles
      )}
      onClick={handleClick}
      disabled={disabled}
    >
      {name}
    </button>
  );
};
