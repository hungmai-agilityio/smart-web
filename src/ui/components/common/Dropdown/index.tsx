'use client';
import { useRef, useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
import { useOutsideClick } from '@/hooks/useOutsideClick';

export interface ItemProps {
  label: string;
  value: string;
}

interface DropdownProps {
  items: ItemProps[];
  onSelect?: (value: string) => void;
}

export const Dropdown = ({ items, onSelect }: DropdownProps) => {
  const [isOpen, setIsOpen] = useState(false);
  const [selectedItem, setSelectedItem] = useState<ItemProps>(items[0]);

  const dropdownRef = useRef<HTMLDivElement>(null);

  const toggleDropdown = () => {
    setIsOpen((prev) => !prev);
  };

  const handleItemClick = (item: ItemProps) => {
    setSelectedItem(item);
    setIsOpen(false);
    if (onSelect) onSelect(item.value);
  };

  useOutsideClick({
    ref: dropdownRef,
    handler: () => setIsOpen(false)
  });

  return (
    <div ref={dropdownRef} className="relative inline-block text-left w-36">
      <button
        type="button"
        className="flex items-center justify-between w-full px-3 py-2 border border-primary rounded-full md:text-base text-x-xs uppercase"
        onClick={toggleDropdown}
      >
        {selectedItem.label}
        <FontAwesomeIcon
          icon={faCaretDown}
          className={`ml-2 transition-transform ${isOpen ? 'rotate-180' : 'rotate-0'}`}
        />
      </button>

      <div
        className={`absolute right-0 w-full mt-1 bg-white border border-gray-300 rounded-md shadow-lg z-10 transition-transform duration-300 ease-in-out ${
          isOpen ? 'opacity-100 translate-y-0' : 'opacity-0 translate-y-2'
        }`}
      >
        {items.map((item) => (
          <button
            key={item.value}
            className="w-full text-left px-4 py-2 hover:bg-gray-100 md:text-base text-xs uppercase"
            onClick={handleItemClick.bind(null, item)}
          >
            {item.label}
          </button>
        ))}
      </div>
    </div>
  );
};
