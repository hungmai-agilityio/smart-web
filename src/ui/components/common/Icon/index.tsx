import { clsx } from 'clsx';

// Font Awesome
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface IconProp {
  icon: IconDefinition;
  quantity?: number;
  styles?: string;
}
export const Icon = ({ icon, quantity, styles }: IconProp) => {
  return (
    <div className={clsx('flex gap-3 items-center', styles)}>
      <FontAwesomeIcon icon={icon} />

      <p>{quantity}</p>
    </div>
  );
};
