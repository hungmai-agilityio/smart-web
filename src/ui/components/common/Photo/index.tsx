'use client';
import { ReactNode, useState } from 'react';
import { clsx } from 'clsx';
import Image, { ImageProps } from 'next/image';

interface PhotoProps extends ImageProps {
  children?: ReactNode;
  styles?: string;
  isCircle?: boolean;
  fixedSize?: { width: number; height: number };
  autoSize?: boolean;
}

export const Photo = ({
  src,
  alt,
  children,
  styles,
  isCircle,
  fixedSize,
  autoSize,
  ...res
}: PhotoProps) => {
  const [hovered, setHovered] = useState(false);

  const toggleHover = () => setHovered(!hovered);

  const containerStyles = clsx('relative', styles, {
    'rounded-full overflow-hidden border-4 border-gray-300 hover:border-second':
      isCircle,
    'w-full h-full': autoSize,
    [`w-[${fixedSize?.width}px] h-[${fixedSize?.height}px]`]:
      fixedSize && !autoSize
  });

  return (
    <div
      className={containerStyles}
      onMouseEnter={toggleHover}
      onMouseLeave={toggleHover}
    >
      <Image
        src={src}
        alt={alt}
        fill={autoSize}
        width={fixedSize?.width}
        height={fixedSize?.height}
        style={{ objectFit: 'cover' }}
        className={clsx('transition-opacity duration-300', {
          'opacity-50': hovered && children
        })}
        {...res}
      />
      {hovered && children && (
        <div className="absolute inset-0 flex items-center bg-black bg-opacity-50 text-white">
          {children}
        </div>
      )}
    </div>
  );
};
