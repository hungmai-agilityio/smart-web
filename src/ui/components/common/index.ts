export * from '@/ui/components/common/Button';
export * from '@/ui/components/common/Dropdown';
export * from '@/ui/components/common/Icon';
export * from '@/ui/components/common/Photo';
export * from '@/ui/components/common/Text';
export * from '@/ui/components/common/StatsRow';
