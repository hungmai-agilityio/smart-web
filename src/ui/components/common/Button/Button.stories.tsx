import { Meta, StoryObj } from '@storybook/react';

// Components
import { Button } from '@/ui/components/common/Button';

// Constants
import { TYPE } from '@/constants';

const meta: Meta<typeof Button> = {
  component: Button,
  tags: ['autodocs'],
  argTypes: {
    name: {
      description: 'Name button'
    },
    variant: {
      description: 'Set type optional for Button'
    },
    onClick: {
      description: 'Handle the event when clicking on the button'
    },
    styles: {
      description: 'Add custom css'
    },
    disabled: {
      description: 'Set disabled button'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    name: 'Submit',
    variant: TYPE.PRIMARY,
    onClick: () => alert('submit')
  }
};

export const Secondary: Story = {
  args: {
    name: 'Submit',
    variant: TYPE.SECOND,
    onClick: () => alert('submit')
  }
};

export const ButtonDisabled: Story = {
  args: {
    name: 'Send',
    disabled: true
  }
};
