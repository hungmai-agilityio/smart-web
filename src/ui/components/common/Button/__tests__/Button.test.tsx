import { render, screen, fireEvent } from '@testing-library/react';

// Constants
import { SIZE, TYPE } from '@/constants';

// Components
import { Button } from '@/ui/components/common/Button';

describe('Button', () => {
  it('renders button with default props', () => {
    render(<Button name="Test Button" />);

    const buttonElement = screen.getByText(/Test Button/i);
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveClass('text-primary');
    expect(buttonElement).not.toBeDisabled();
  });

  it('renders button with size prop', () => {
    render(<Button name="Test Button" />);

    const buttonElement = screen.getByText(/Test Button/i);
    expect(buttonElement).toHaveClass('px-2 rounded-full font-light');
  });

  it('renders button with variant prop', () => {
    render(<Button name="Test Button" variant={TYPE.SECOND} />);

    const buttonElement = screen.getByText(/Test Button/i);
    expect(buttonElement).toHaveClass('bg-white');
    expect(buttonElement).toHaveClass('border-primary');
  });

  it('button is disabled when disabled prop is true', () => {
    render(<Button name="Test Button" disabled />);

    const buttonElement = screen.getByText(/Test Button/i);
    expect(buttonElement).toBeDisabled();
    expect(buttonElement).toHaveClass('cursor-not-allowed');
  });

  it('button click triggers alert', () => {
    const alertMock = jest.spyOn(window, 'alert').mockImplementation();
    render(<Button name="Test Button" />);

    const buttonElement = screen.getByText(/Test Button/i);
    fireEvent.click(buttonElement);

    expect(alertMock).toHaveBeenCalledWith('We will update feature later');
    alertMock.mockRestore();
  });

  it('should be render match to snapshot', () => {
    const { container } = render(
      <Button name="Click me" variant={TYPE.PRIMARY} />
    );

    expect(container).toMatchSnapshot();
  });
});
