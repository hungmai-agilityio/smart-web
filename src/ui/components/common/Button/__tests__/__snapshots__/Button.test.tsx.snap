// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be render match to snapshot 1`] = `
<div>
  <button
    class="text-primary px-2 rounded-full font-light truncate w-24 h-6 text-xs sm:w-28 sm:h-8 sm:text-sm md:w-36 md:h-10 md:text-base lg:w-52 lg:h-12 lg:text-lg bg-second"
  >
    Click me
  </button>
</div>
`;
